### PR TITLE
Change Get-Content $Path -Encoding byte to Get-Content $Path -AsByteS…

### PR DESCRIPTION
**Pull request by tonyping**
…tream for compatibility with Powershell Core

### CODE REVIEW

**Comment by vexx32** [comment]
Given that the newer syntax is completely incompatible with older versions of PowerShell (which choco still supports), we'd need to manage this with an if/else based on the PowerShell version; we can't simply use newer syntaxes at the present without making sure older versions will still work.

**Comment by tonyping** [comment]
Ah sorry, I didn't know that. So would doing a simple version check and then selecting which parameter would work (-Encoding Byte for older, or -AsByteStream for newer) be enough, or are there any further consideration that Powershell Core needs to be fully compatible with the installer? 

Because I saw this[issue](https://github.com/chocolatey/choco/issues/2083), that mentions that Core does not have OS environment variables, but I'm not really sure what the replier meant by that.

**Comment by vexx32** [comment]
Yeah I've no idea what they mean by that, but I think they misunderstood the issue / underlying problem.

I would... hmm. Yeah, probably a version gated check for anything less than 6 should use the old form and otherwise use the new switch is probably the safest bet. 🙂

**Comment by gep13** [comment]
@vexx32 are you in a position to re-review this?

@tonyping was an issue created that this PR can be linked to?

File: nuget/chocolatey/tools/chocolateysetup.psm1 [hunk 1]
```diff
@@ -549,7 +549,14 @@ param(
...
+  if ($PSVersionTable.PSVersion.Major -lt 6) {
+    Write-Debug "Detected Powershell version < 6 ; Using -Encoding byte parameter"
     $bytes = [byte[]](Get-Content $Path -Encoding byte -ReadCount 4 -TotalCount 4)
+  }
+  else {
+    Write-Debug "Detected Powershell version >= 6 ; Using -AsByteStream parameter"
+    $bytes = [byte[]](Get-Content $Path -AsByteStream -ReadCount 4 -TotalCount 4)
+    }
 
     if(!$bytes) { return 'utf8' }
```
Comment (vexx32) [inline nuget/chocolatey/tools/chocolateysetup.psm1:561]: This looks good to me, apart from the indentation seemingly not matching the rest of the function here. Can you bump the indentation so it matches the rest of this function?

**Review by vexx32 — APPROVED** [review]
Looks good to me with one minor formatting tweak.

**Comment by gep13** [comment]
I am going to close this PR, as this has been updated in a new PR here:

https://github.com/chocolatey/choco/pull/2416

Let's use that one going forward.  Correct attribution will still be applied in the git commits that are merged into the develop branch.

**Pull request closed without merging**